libbeat/docs/shared-env-vars.asciidoc

```diff
@@ -49,3 +65,38 @@ and what each configuration looks like after replacement:
 |`name: ${NAME:beats}` |no setting            |`name: beats`
 |`name: ${NAME:beats}` |`export NAME=elastic` |`name: elastic`
 |==================================
```

Comment:
@urso I think we should document the `:?` operator and adjust the go-ucfg message. It seems likely that some users will want to return an error that provides some context when a variable cannot be resolved.